File: fontes/backend/c-sharp/T2TiRetaguardaSH/T2TiRetaguardaSH/Controllers/Cadastros/ErpTipoPlanoController.cs
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using T2TiRetaguardaSH.Models;
using T2TiRetaguardaSH.Services;

namespace T2TiRetaguardaSH.Controllers
{
    [Route("erp-tipo-plano")]
    [Produces("application/json")]
    public class ErpTipoPlanoController : Controller
    {
		private readonly ErpTipoPlanoService _service;

        public ErpTipoPlanoController()
        {
            _service = new ErpTipoPlanoService();
        }

        [HttpGet]
        public IActionResult GetListErpTipoPlano([FromQuery]string filter)
        {
            try
            {
                IEnumerable<ErpTipoPlanoModel> resultList;
                if (filter == null)
                {
                    resultList = _service.GetList();
                }
                else
                {
                    // defines filter
                    Filtro filterObj = new Filtro(filter);
                    resultList = _service.GetListFilter(filterObj);
                }
                return Ok(resultList);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Error [GetList ErpTipoPlano]", ex));
            }
        }

        [HttpGet("{id}", Name = "GetObjectErpTipoPlano")]
        public IActionResult GetObjectErpTipoPlano(int id)
        {
            try
            {
                var obj = _service.GetObject(id);

                if (obj == null)
                {
                    return StatusCode(404, new RetornoJsonErro(404, "Not Found [GetObject ErpTipoPlano]", null));
                }
                else
                {
                    return Ok(obj);
                }
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Error [GetObject ErpTipoPlano]", ex));
            }
        }

        [HttpPost]
        public IActionResult InsertErpTipoPlano([FromBody]ErpTipoPlanoModel objJson)
        {
            try
            {
                if (!ModelState.IsValid)
                {
                    return StatusCode(400, new RetornoJsonErro(400, "Invalid Object [Insert ErpTipoPlano]", null));
                }
                _service.Insert(objJson);

                return CreatedAtRoute("GetObjectErpTipoPlano", new { id = objJson.Id }, objJson);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Error [Insert ErpTipoPlano]", ex));
            }
        }

        [HttpPut]
        public IActionResult UpdateErpTipoPlano([FromBody]ErpTipoPlanoModel objJson, int id)
        {
            try
            {
                if (!ModelState.IsValid)
                {
                    return StatusCode(400, new RetornoJsonErro(400, "Invalid Object [Update ErpTipoPlano]", null));
                }

                _service.Update(objJson);

                return GetObjectErpTipoPlano(id);
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Error [Update ErpTipoPlano]", ex));
            }
        }

        [HttpDelete("{id}")]
        public IActionResult DeleteErpTipoPlano(int id)
        {
            try
            {
                var obj = _service.GetObject(id);

                _service.Delete(obj);

                return Ok();
            }
            catch (Exception ex)
            {
                return StatusCode(500, new RetornoJsonErro(500, "Error [Delete ErpTipoPlano]", ex));
            }
        }

    }
}

File: fontes/backend/python/retaguarda_sh/src/controller/erp_tipo_plano_controller.py
from flask import Blueprint, request, jsonify
from src.model.transient.filter import Filter
from src.service.erp_tipo_plano_service import ErpTipoPlanoService

erp_tipo_plano_bp = Blueprint('erp-tipo-plano', __name__)
service = ErpTipoPlanoService()

@erp_tipo_plano_bp.route('/erp-tipo-plano', methods=['GET'])
@erp_tipo_plano_bp.route('/erp-tipo-plano/', methods=['GET'])
def get_list():
    query_params = request.args
    filter_obj = Filter(query_params)    
    if filter_obj.where:
        result_list = service.get_list_filter(filter_obj)
    else:
        result_list = service.get_list()
    return jsonify([obj.serialize() for obj in result_list])

@erp_tipo_plano_bp.route('/erp-tipo-plano/<int:id>', methods=['GET'])
def get_object(id):
    obj = service.get_object(id)
    return jsonify(obj.serialize())

@erp_tipo_plano_bp.route('/erp-tipo-plano', methods=['POST'])
def insert():
    data = request.json
    result = service.insert(data)
    return jsonify(result.serialize()), 201

@erp_tipo_plano_bp.route('/erp-tipo-plano', methods=['PUT'])
def update():
    data = request.json
    result = service.update(data)
    return jsonify(result.serialize()), 200

@erp_tipo_plano_bp.route('/erp-tipo-plano/<int:id>', methods=['DELETE'])
def delete(id):
    service.delete(id)
    return jsonify({'message': 'Deleted successfully'})